joomla/joomla-cms
# Correct db schema update errors for 3.5.0

Pull Request for Issue #9156 .
#### Summary of Changes

This PR changes following:
- Correct handling of alter table statements in the database schema manager for the UTF8MB4 changes
- Extend the database schema manager by handling alter table statements with drop and add index in one statement for the UTF8MB4 changes
- Add missing index modifications to the SQL script for the UTF8MB4 changes
- Add missing index lengths limitations to utf8mb4_bin columns to the joomla installation SQL script.

It **shall not** fix all possible problems on updating from 3.4.8 to 3.5.0.

It **shall** fix only the errors with fixing database problems being reported after the update.
#### Testing Instructions

Either do a Joomla! Update from a 3.4.8 to a "3.5.0 latest staging plus this PR applied":

You can find an update container here:
[http://test5.richard-fath.de/Joomla_3.5.0-beta2-Beta-Update_Package_test1.zip](http://test5.richard-fath.de/Joomla_3.5.0-beta2-Beta-Update_Package_test1.zip).

Or you can use the Joomla! Update component with following custom URL (note it is http, not use https):
[http://test5.richard-fath.de/list_test4.xml](http://test5.richard-fath.de/list_test4.xml).

Or do as @wilsonge described in his issue #9156, i.e.
- Install a new, clean 3.4.8 into an empty database,
- download the branch of this patch here [https://github.com/richard67/joomla-cms/archive/correct-db-schema-manager-1.zip](https://github.com/richard67/joomla-cms/archive/correct-db-schema-manager-1.zip),
- unzip it into the 3.4.8 installation, so all the files are updated now by the ones from this PR,
- login at the backend and check for database problems (Extenions -> Manage -> Database).
- You will see one or more problems reported, depending on UTF8MB4 of your database server and client.
- Use the "Fix" button.
- All database problems should be fixed.
